File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/SBPacketAuth.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;
import pize.tests.voxelgame.server.net.ServerLoginPacketHandler;

import java.io.IOException;

public class SBPacketAuth extends IPacket<ServerLoginPacketHandler>{
    
    public static final int PACKET_ID = 1;
    
    public SBPacketAuth(){
        super(PACKET_ID);
    }
    
    
    public String accountSessionToken;
    
    public SBPacketAuth(String accountSessionToken){
        this();
        this.accountSessionToken = accountSessionToken;
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeUTF(accountSessionToken);
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        accountSessionToken = stream.readUTF();
    }
    
    @Override
    public void handle(ServerLoginPacketHandler packetListener){
        packetListener.handleAuth(this);
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/SBPacketAuth.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.tcp.packet.IPacket
import pize.tests.voxelgame.server.net.ServerLoginPacketHandler
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException

class SBPacketAuth() : IPacket<ServerLoginPacketHandler?>(PACKET_ID) {
    var accountSessionToken: String? = null

    constructor(accountSessionToken: String?) : this() {
        this.accountSessionToken = accountSessionToken
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeUTF(accountSessionToken)
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        accountSessionToken = stream!!.readUTF()
    }

    override fun handle(packetListener: ServerLoginPacketHandler) {
        packetListener.handleAuth(this)
    }

    companion object {
        const val PACKET_ID = 1
    }
}
